### ISSUE TITLE
MSE Loss with Inequalities Denominator

### ISSUE THREAD

**Issue by kaboroevich** [openvax/mhcflurry#220]
Can someone give further explanation of the calculation of the [denominator in the custom loss function](https://github.com/openvax/mhcflurry/blob/6dc760b75f9ae78a6c7ea28bd3ed2d13faaf1c8d/mhcflurry/custom_loss.py#L191)? 

From the explanation, I assumed it would either be the number of all measurements or be the sum of:
1. the number of all the (=) equality measurements, 
2. the number of (>) inequality measurements with a prediction less than the truth (difference less than zero), and 
3. the number of (<) inequality measurements with a prediction greater than the truth (difference greater than zero). 

In other words, the number of all predictions that contribute to the numerator.

However, from the formula, it's the number of all measurements less the number of (>) equalities that have a truth value of zero. Why is this specific subset of measurements not counted? 

If this is explained in either paper and I overlooked it, I apologize.

**Comment by timodonnell** [comment]
Good question. It's an implementation thing, not mentioned in the paper. The ">0" measurements aren't real data.  They are added as padding values in some routine. They never contribute to loss since they are always satisfied by any prediction, since predictions are in (0, 1). So we don't count them in the denominator as they never contribute to the numerator.

**Comment by kaboroevich** [comment]
@timodonnell thank you very much for that explanation.
